CRM/Price/BAO/LineItem.php

```diff
@@ -444,6 +444,12 @@ public static function processPriceSet($entityId, $lineItem, $contributionDetail
           if ($line['entity_table'] == 'civicrm_contribution') {
             $line['entity_id'] = $contributionDetails->id;
           }
+          // CRM-19094: entity_table is set to civicrm_membership then ensure
+          // the entityId is set to membership ID not contribution by default
+          elseif ($line['entity_table'] == 'civicrm_membership' && !empty($line['entity_id']) && $line['entity_id'] == $contributionDetails->id) {
```

Comment:
@JoeMurray I agree its a hackish fix for a bug which arises when you got more number of membership then contribution. In fact I know the exact spot https://github.com/civicrm/civicrm-core/blob/master/CRM/Member/Form/Membership.php#L1586 where we are passing contribution ID against membership table. Also earlier I tried to make a generic fix to ensure that we always pass the correct entity ID for corresponding entity table but then lots of unit test were failing ~28 in count. So for now this fix will do, as it is not changing any value BUT set proper membership ID on edit if its incorrect. I will surely raise a issue against 4.7.11 to ensure that we always pass correct entity id against entity table.